docs/debugging-transactions.md

```diff
@@ -0,0 +1,510 @@
+
+### Example of Debugging a Transaction
+
+Here’s a simple example of how to deploy and interact with a contract in Remix:
+
+#### Solidity Contract
+
+```solidity
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity ^0.8.0;
+
+contract SimpleStorage {
+    uint256 public storedData;
+
+    function set(uint256 x) public {
+        storedData = x;
+    }
+
+    function get() public view returns (uint256) {
+        return storedData;
+    }
+}
+```
+
+#### Steps in Remix
+
+1. **Deploy the Contract**:
+    - Compile the `SimpleStorage` contract.
+    - Deploy it using the "Deploy & Run Transactions" tab.
+
+2. **Interact with the Contract**:
+    - Call the `set` function with a value (e.g., `set(42)`).
+
+3. **Debug the Transaction**:
+    - Go to the terminal in Remix to see the transaction details.
+    - Find the `set` transaction and click the debug icon.
+    - Use the debugger to step through the transaction and inspect the state changes.
+
+## 2. Debugging with Truffle
+
+[Truffle](https://archive.trufflesuite.com) is a popular development framework for Ethereum smart contracts that
+provides
+built-in debugging capabilities.
+
+### Steps to Use the Debugger in Truffle
+
+Official Documentation:
+[Truffle Debugger](https://archive.trufflesuite.com/docs/truffle/how-to/debug-test/use-the-truffle-debugger/)
+
+1. **Install Truffle**:
+    - Install Truffle using npm: `npm install -g truffle`.
+2. **Initialize a new Truffle project**:
+    - Initialize using `truffle init`.
+3. **Set Up Your Environment**:
+    - Install the necessary dependencies: `npm install @truffle/hdwallet-provider`
+    - Configure your development environment in the `truffle-config.js` file.
+    - Ensure you have a local node running that supports the `debug_traceTransaction` method.
+
+- Connect to the local node by specifying the network configuration. (e.g., `http://localhost:7546`)
+  ```javascript
+  const HDWalletProvider = require('@truffle/hdwallet-provider');
+  module.exports = {
+    networks: {
+      development: {
+        host: "127.0.0.1",
+        port: 7546,
+        network_id: '*',
+        provider: () => new HDWalletProvider(
+          [
+            '0x105d050185ccb907fba04dd92d8de9e32c18305e097ab41dadda21489a211524', // operator private key
+            '0x2e1d968b041d84dd120a5860cee60cd83f9374ef527ca86996317ada3d0d03e7' // receiver private key
+          ],
+          'http://localhost:7546'
+        )
+      },
+    },
+  };
+  ```
+
+1. **Write and Compile Your Contracts**:
+    - Add your `SimpleStorage` contract in the `contracts/` directory.
+    - Compile your contracts: `truffle compile`.
+2. **Deploy and Interact with Your Contracts**:
+    - Deploy your contracts to the local network: `truffle migrate`.
+    - Interact with your contracts using Truffle console or scripts.
+    - Call functions and send transactions to test your contracts.
```

Comment:
Also here, would screenshot(s) help?

Reply:
I will add an explanation on how to do it through Metamask or a ethers/web3js script